beam.py

```diff
@@ -135,23 +159,43 @@ def authenticate(self, *args):
 
             self.send_message(*args[:-1], method="auth")
 
+            if self.quiet:
+                self.http_session = Session()
+
             self.read_chat(self.handle)
         else:
-            raise ConnectionError(future.exception())
+            self.logger.error("There was an issue connecting. Trying again")
+            self.authenticate(*args)
```

Comment:
Technically, this has the potential to raise a `RecursionError` - it calls itself, so if it fails to authenticate enough times (997, I believe), it will crash. Since `v0.4` fixes this, though, it's not a huge deal to leave it.